e2e/basic.spec.ts

```diff
@@ -0,0 +1,27 @@
+import {test, expect} from '@playwright/test';
+
+[
+  ['iframe', 'vanilla'],
+  ['iframe', 'preact'],
+  ['iframe', 'svelte'],
+  ['iframe', 'vue'],
+  ['iframe', 'htm'],
+].forEach(([sandbox, example]) => {
+  test(`basic modal interaction with ${sandbox} sandbox and ${example} example`, async ({
+    page,
+  }) => {
+    await page.goto(`/?sandbox=${sandbox}&example=${example}`);
+
+    await page.getByRole('button', {name: 'Open modal'}).click();
+    await page.getByRole('button', {name: 'Click me!'}).click();
+    await page.getByRole('button', {name: 'Click me!'}).click();
+
+    await expect(page.getByText('Click Count: 2')).toBeVisible();
+
+    page.once('dialog', (dialog) => {
```

Comment:
Can we store this as a promise and make sure it resolves, to ensure that the contained `expect()` is actually hit?